File: Data Structures & Algorithms/is-palindrome/submission-0.js
class Solution {
    /**
     * @param {string} s
     * @return {boolean}
     */
    isPalindrome(s) {
        const cleaned = s.toLowerCase().replace(/[^a-z0-9]/g, '');
        let left = 0
        let right = cleaned.length - 1

        while(left < right){
            if(cleaned[left] !== cleaned[right]){
                return false
            }
            
            left++
            right--
        }
        return true
    }
}

File: Data Structures & Algorithms/generate-parentheses/submission-0.py
class Solution:
    def generateParenthesis(self, n: int) -> List[List[str]]:
        res = []

        def backtrack(current, left, right):
            if len(current) == n * 2:
                res.append(current)
                return

            if left < n:
                backtrack(current + "(", left + 1, right)
            
            if left > right:
                backtrack(current + ")", left, right + 1)

        backtrack("", 0, 0)

        return res
